frontend/public/components/utils/firehose.jsx

```diff
@@ -161,17 +165,28 @@ export const Firehose = connect(
       this.clear();
     }
 
-    shouldComponentUpdate(nextProps) {
+    shouldComponentUpdate(nextProps, nextState) {
       const { forceUpdate = false } = this.props;
-      if (nextProps.inFlight !== this.props.inFlight && nextProps.loaded) {
+      if (
+        this.state === nextState &&
+        nextProps.inFlight !== this.props.inFlight &&
+        nextProps.loaded
+      ) {
         return forceUpdate;
       }
+      if (
+        Object.keys(nextProps).every((key) => nextProps[key] === this.props[key]) &&
+        nextState.firehoses.length === 0 &&
+        this.state.firehoses.length === 0
+      ) {
+        return false;
```

Comment:
Should this be `return forceUpdate`? We might add an early return at the top of the function when `forceUpdate` is true.

Reply:
i guess you're right, but can you explain the purpose of `forceUpdate` ? It is unclear to me..

Reply:
It was a hack because firehose was not always updating :(

It's possible we can remove it with this fix. We needed it specifically for the workloads tab on the projects page. @TheRealJon might remember exactly what scenario caused problems.